components/pagination/stories/template.js

```diff
@@ -0,0 +1,88 @@
+import { html } from "lit-html";
+import { classMap } from "lit-html/directives/class-map.js";
+import { repeat } from 'lit-html/directives/repeat.js';
+
+import { Template as Button } from "@spectrum-css/button/stories/template.js";
+import { Template as ActionButton } from "@spectrum-css/actionbutton/stories/template.js";
+import { Template as Textfield } from "@spectrum-css/textfield/stories/template.js";
+
+
+import '../index.css';
+import '../skin.css';
+
+// More on component templates: https://storybook.js.org/docs/web-components/writing-stories/introduction#using-args
+export const Template = ({
+  rootClass = "spectrum-Pagination",
+  size = "m",
+  customClasses = [],
+  variant,
+  items,
+  ...globals
+}) => {
+
+  if (variant === 'explicit') {
+    return html`
+      <nav class=${classMap({
+        [rootClass]: true,
+        [`${rootClass}--${variant}`]: typeof variant !== "undefined",
+        ...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
+      })}
+      >
+        ${ActionButton({
+          size,
+          isQuiet: true,
+          icon: "ChevronLeft100",
+          customClasses: [`${rootClass}-prevButton`]
+        })}
+        ${Textfield({
+          value: "1",
+          customClasses: [`${rootClass}-textfield`]
+        })}
+        <span class="spectrum-Body--secondary ${rootClass}-counter">of 89 pages</span>
+        ${ActionButton({
+          size,
+          isQuiet: true,
+          icon: "ChevronRight100",
+          customClasses: [`${rootClass}-nextButton`]
+        })}
+      </nav>
+    `;
+  }
+  return html`
+    <nav class=${classMap({
+      [rootClass]: true,
+      [`${rootClass}--${variant}`]: typeof variant !== "undefined",
+      ...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
+    })}
+    >
+      ${Button({
+        size,
+        variant: "primary",
+        treatment: "outline",
+        label: "Prev",
+        customClasses: [`${rootClass}-prevButton`],
+        ...globals
+      })}
+      ${repeat(items, (item) => item.id, (item) => {
+        if (typeof item === "object") {
+          return ActionButton({
+            ...globals,
+            ...item,
+            isQuiet: true,
+          })
+        } else return item;
+      })}
+
+      ${Button({
+        size,
+        variant: "primary",
+        treatment: "outline",
+        label: "Next",
+        customClasses: [`${rootClass}-nextButton`],
+        ...globals
```

Comment:
I'd recommend passing globals as a first variable because it's a spread and we can't be entirely sure what it contains. On the off chance it contains a conflicting variable name with the ones you list above, we want your hardcoded versions to "win".